src/gui/dialog/LatexSettingsPanel.cpp

```diff
@@ -0,0 +1,38 @@
+#include "LatexSettingsPanel.h"
+
+LatexSettingsPanel::LatexSettingsPanel(GladeSearchpath* gladeSearchPath):
+        GladeGui(gladeSearchPath, "latexSettings.glade", "latexSettingsPanel"),
+        cbAutoDepCheck(GTK_TOGGLE_BUTTON(this->get("latexSettingsRunCheck"))),
+        globalTemplateChooser(GTK_FILE_CHOOSER(this->get("latexSettingsTemplateFile"))) {
+    g_object_ref(this->cbAutoDepCheck);
+    g_object_ref(this->globalTemplateChooser);
+}
+
+LatexSettingsPanel::~LatexSettingsPanel() {
+    g_object_unref(this->cbAutoDepCheck);
+    g_object_unref(this->globalTemplateChooser);
+}
+
+void LatexSettingsPanel::load(const LatexSettings& settings) {
+    gtk_toggle_button_set_active(this->cbAutoDepCheck, settings.autoCheckDependencies);
+    if (!settings.globalTemplatePath.empty()) {
+        gtk_file_chooser_set_filename(this->globalTemplateChooser, settings.globalTemplatePath.c_str());
```

Comment:
Should be `settings.globalTemplatePath.u8string().c_str()` since glib assumes UTF-8 encoded strings.